### PR TITLE
chore: improvement azureaisearch limits

### PR DESCRIPTION
**Pull request by EmanuelCampos**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py [hunk 1]
```diff
@@ -713,15 +713,15 @@ def add(
         ids = []
         accumulated_size = 0
         max_size = 16 * 1024 * 1024  # 16MB in bytes
-        max_docs = 1000
+        max_docs = 600  # the limit is 1000 reduced to 600 to avoid the 16MB limit
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py:716]: Why do we have a max docs parameter? All that matters is the max size right? We could just set the max size to like 10-14MB

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py [hunk 2]
```diff
@@ -713,15 +713,15 @@ def add(
...
 
         for node in nodes:
             logger.debug(f"Processing embedding: {node.node_id}")
             ids.append(node.node_id)
 
             index_document = self._create_index_document(node)
             document_size = len(
-                str(node.get_content(metadata_mode=MetadataMode.NONE)).encode("utf-8")
+                str(node.get_content(metadata_mode=MetadataMode.ALL)).encode("utf-8")
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py:724]: Should we be measuring the size of index_document instead?

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py (hunk 1 above)
Comment (EmanuelCampos) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py:716]: @logan-markewich the api also limits to 1000 documents/nodes at time, not just by file size

File: llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py (hunk 2 above)
Comment (EmanuelCampos) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py:724]: will change it

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**